packages/app/obojobo-repository/shared/components/notification.test.js

```diff
@@ -0,0 +1,151 @@
+		expect(tree).toMatchSnapshot()
+	})
+
+	test('loads notifications from cookies on mount', () => {
+		const onDataFromNotification = jest.fn()
+		const notificationValue = [{ key: 1, text: 'Test Notification', title: 'Test Title' }]
+		document.cookie = `notifications=${JSON.stringify(notificationValue)}`
+
+		const component = create(<Notification onDataFromNotification={onDataFromNotification} />)
+		const tree = component.toJSON()
+
+		expect(tree).toMatchSnapshot()
+		expect(document.cookie).toBe(
+			`notifications=${encodeURIComponent(JSON.stringify(notificationValue))};`
+		)
+	})
+
+	test('handles click on exit button and updates state and cookie', () => {
+		const onDataFromNotification = jest.fn()
+		const notificationValue = [
+			{ key: 1, text: 'Notification1', title: 'Title1' },
+			{ key: 2, text: 'Notification2', title: 'Title2' }
+		]
+		document.cookie = `notifications=${JSON.stringify(notificationValue)}`
+
+		const reusableComponent = <Notification onDataFromNotification={onDataFromNotification} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+
+		let tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		// Simulate click on exit button for the first notification
+		const key = 0
+		const exitButtons = component.root.findAllByProps({ className: 'notification-exit-button' })
+
+		if (exitButtons[key]) {
+			act(() => {
+				exitButtons[key].props.onClick()
+			})
+
+			tree = component.toJSON()
+			expect(tree).toMatchSnapshot()
+			expect(document.cookie).toBe(
+				`notifications=${encodeURIComponent(JSON.stringify(notificationValue))};`
+			)
+		}
+	})
+
+	test('hides the notification on exit button click', () => {
+		const onDataFromNotification = jest.fn()
+		document.cookie =
+			'notifications=' +
+			JSON.stringify([{ key: 1, text: 'Test Notification', title: 'Test Title' }])
+
+		const elementToExit = document.getElementsByClassName('notification-exit-button')[0]
+		const reusableComponent = <Notification onDataFromNotification={onDataFromNotification} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+
+		let tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		const key = 0
+		const exitButtons = component.root.findAllByProps({ className: 'notification-exit-button' })
+
+		act(() => {
+			exitButtons[key].props.onClick()
+		})
+
+		tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		if (elementToExit) {
+			expect(elementToExit.style.display).toBe('undefined')
+		}
+	})
+	test('renders null when there are no notifications but document.cookie is not null', () => {
+		const onDataFromNotification = jest.fn()
+		const reusableComponent = <Notification onDataFromNotification={onDataFromNotification} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+		const tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		if (document && document.cookie) {
```

Comment:
Since this test is only being run one time and should only be testing one thing, there should probably not be an `if...else` here to determine which expectations are run. In this case you could replace this `if...else` with two expectations:
`expect(document).toBeTruthy()` and `expect(document.cookie).toBeTruthy()`. At that point you can reasonably assume that only the `if` block would ever run, so you wouldn't need the `else` block or the `if...else` at all.

Reply:
oops i totally missed this but rewrote the very similar tests yesterday. Fixing now